### DISCUSSION TITLE
Focusing

### DISCUSSION THREAD

**Discussion by pw310** [aaronwmorris/indi-allsky#1445] [Q&A]
Am I missing something glaringly obvious?

I have the Pi camera CM3, imx708. I have put it in focus mode, but can't find where I can make any adjustment to the focus. I have a refresh rate, zoom and z & y offset and an untitled graph, which I presume is the score, which fluctuates between 7 and 15. No focus control.

So far I have been using the device to generate time-lapses of my back garden, until I get round to sorting out an enclosure. The camera was in focus from the start, however the pi lost connectivity for whatever reason and I had to reboot it by removing the power. Now it is no longer in focus.

**Comment by aaronwmorris** [comment]
The focus view does not have any method of controlling the focus of the Pi cameras.  The focus is more for manual focusing with a lens or electronic focuser.

Try adding these parameters to the libcamera extra commands (both fields) on the Camera config tab and reload/restart.  Lens position 0.0 is supposed to be infinity.

```
--autofocus-mode manual --lens-position 0.0
```

https://www.raspberrypi.com/documentation/computers/camera_software.html#lens-position

**Comment by pw310** [comment]
I updated Night libcamera extra options and Day libcamera extra options on the Camera config tab, unfortunately that's not made any difference to the focus, but the status is now "DOWN" in red, but it's still taking out of focus pictures. and displaying them in the Latest view.

**Reply by aaronwmorris** [reply]
Have you tried completely powering your system down, disconnect power, to try to reinitialize the camera?

**Reply by pw310** [reply]
It needed 2 power downs for some reason, as I was unable to ssh to it on the first boot. Looks like it is creeping back into focus over several frames though, which is a good sign.

Thanks for your help.

**Reply by pw310** [reply]
Hello again, I have a second camera, this one is in a weatherproof enclosure and the camera is protected by Entaniya Waterproof Case for Raspberry Pi Camera Modules. I have put it in Focus mode waited hours for it to focus, but it never achieves focus. I have tried to chage the lens position:

--autofocus-mode manual --lens-position 0.0
--autofocus-mode manual --lens-position 0.1
--autofocus-mode manual --lens-position 0.2
--autofocus-mode manual --lens-position 10
--autofocus-mode manual --lens-position 20
--autofocus-mode manual --lens-position 30
--autofocus-mode manual --lens-position 300
--autofocus-mode manual --lens-position -300
--autofocus-mode manual --lens-position -10
--autofocus-mode manual --lens-position -20
--autofocus-mode manual --lens-position -0.5

there is no change in the size of the star I am capturing, the attach pic is of Vega, on high magnification
![image](https://github.com/user-attachments/assets/caad6f51-291a-4c09-a2b9-56e17cb5ef2c)


Does it matter that the camera is pointing directly up?

**Reply by pw310** [reply]
A further update, I have spent a few hours today taking frames with different lens positions, thankfully it wasn't clear and managed to use white fluffy clouds to determine an ideal lens position of approx 6.

Once dark, I have restarted indi-allsky with a lens position of 6 and the shape of Vega remains the same, which leads me to believe that this lens cover, although ideally suited to this IMC708 camera, should be limited to use in fish tanks, or outdoor weatherproofed coverage and is not optically suited for astronomy. If that is the case, if you have any recommendations for camera dome suppliers that are optically sufficient for Astrophotography wide field applications, I would be interested to hear them.
